tools/storage/uri/src/test/java/org/projectnessie/storage/uri/TestStorageUri.java

```diff
@@ -0,0 +1,212 @@
+    "s3://b/path,s3://b/path",
+    "s3://b/path,s3://b/path/",
+    "s3://b/path,s3://b/path//",
+    "s3://b/path,/path",
+  })
+  void testCompare(String v1, String v2) {
+    StorageUri s1 = StorageUri.of(v1);
+    StorageUri s2 = StorageUri.of(v2);
+    URI u1 = normalizedURI(v1);
+    URI u2 = normalizedURI(v2);
+    assertThat(s1.equals(s2)).isEqualTo(u1.equals(u2));
+    assertThat(s2.equals(s1)).isEqualTo(u2.equals(u1));
+    assertThat(Integer.signum(s2.compareTo(s1))).isEqualTo(Integer.signum(u2.compareTo(u1)));
+    assertThat(Integer.signum(s1.compareTo(s2))).isEqualTo(Integer.signum(u1.compareTo(u2)));
+  }
+
+  @ParameterizedTest
+  @ValueSource(
+      strings = {
+        "/path/file",
+        "relative/file",
+        "file:/",
+        "file:/path",
+        "file:/path/file",
+        "file:/path/file//",
+        "file:///path/file/",
+        "file:///path/file",
+        "file://///path/file",
+        "file:///path/file//",
+        "file:///path////file//",
+        "s3:/path",
+        "s3://bucket",
+        "s3://bucket/",
+      })
+  void testLocation(String input) {
+    StorageUri s = StorageUri.of(input);
+    URI u = normalizedURI(input);
+    assertThat(s.location()).isEqualTo(u.toString());
+    assertThat(s.toString()).isEqualTo(u.toString());
+  }
+
+  @ParameterizedTest
+  @CsvSource({
+    "s3://bucket/\"file,s3://bucket/\"file",
+    "s3://bucket/\000file,s3://bucket/\000file",
+  })
+  void testLocationSpecialChars(String input, String expected) {
+    assertThat(StorageUri.of(input).location()).isEqualTo(expected);
+    assertThat(StorageUri.of(input).toString()).isEqualTo(expected);
+  }
+
+  @Test
+  void testLocalPath() {
+    Path local = Paths.get("/path/file");
+    assertThat(StorageUri.of("file:/path/file").toLocalPath()).isEqualTo(local);
+    assertThat(StorageUri.of("file:///path/file").toLocalPath()).isEqualTo(local);
+
+    assertThatThrownBy(() -> StorageUri.of("/unspecified").toLocalPath())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Not a local file: /unspecified");
+    assertThatThrownBy(() -> StorageUri.of("s3://bucket/path").toLocalPath())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Not a local file: s3://bucket/path");
+  }
+
+  @ParameterizedTest
+  @CsvSource({
+    "/base,/base/file,file",
+    "/base/,/base/file,file",
+    "/base/,/base//file,file",
+    "/base,/base////file,file",
+    "/ba,/base////file,file",
+    "file:///base/,file:/base/file",
+    "file:///base,file:/base/file",
+    "file:/ba,file:/base/file",
+    "file:opaque/a,path",
+    "file:opaque/a,file:path",
+    "file:opaque,file:/path",
+    "file:opaque/a,file:/path",
+    "s3://b/path,s3://b/path/file,file",
+    "s3://b/path/,s3://b/path/file,file",
+    "s3://b/path/,s3://b///path/file,file",
+    "s3://b/,s3://b/path/file,path/file",
+    "s3://b,s3://b/path/file,path/file",
+  })
+  void testRelativize(String base, String other) {
```

Comment:
The third parameter in the CSVs is unused??

Reply:
fixed